### PR TITLE
Reader: Add a way to view posts that are restricted

### LINKED ISSUE #23638: Post details inconsistently displayed with [...]
### Expected behavior
When clicking the post details, I should see either the post or the cropped version consistently. If cropped version appears, there should be a link to the full version.

### Actual behavior
I sometimes see a cropped post ending with [...], and sometimes see the real one. The cropped one has no link to the full one.

### Steps to reproduce the behavior
1. Subscribe to the site https://mydelarosa.com/
2. In the reader, click a post - cropped version appears. 
3. Cropped version has no link to the full version
4. Then click the site, and click the post again - full version appears

https://github.com/user-attachments/assets/35185809-854b-4be7-8ffe-7e87942a0389

### PR DESCRIPTION
**Pull request by kean**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/23638 by adding support for posts that can't be viewed in the Reader.

Screenshot 2024-11-21 at 4 03 40 PM
Screenshot 2024-11-21 at 4 03 43 PM

https://github.com/user-attachments/assets/26d81fbf-f727-411a-86ce-01e4c323b5ba

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift [hunk 1]
```diff
@@ -247,6 +247,12 @@ class ReaderDetailViewController: UIViewController, ReaderDetailView {
 
         webView.isP2 = post.isP2Type()
 
+        if post.content?.hasSuffix("[…]") == true {
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift:250]: There gotta be a better way to test, but I couldn't find one. I think it'll do for now.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderReadMoreView.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        addSubview(visualEffectView)
+        visualEffectView.pinEdges(insets: UIEdgeInsets(top: 80, left: 0, bottom: 16, right: 0))
+
+        textView.adjustsFontForContentSizeCategory = true
+
+        let string = NSMutableAttributedString(string: Strings.viewForMore, attributes: [
+            .font: UIFont.preferredFont(forTextStyle: .body),
+            .foregroundColor: UIColor.label
+        ])
+        let range = string.mutableString.range(of: "%@")
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderReadMoreView.swift:30]: I guess we'd just have to hope the `%@` in the original format doesn't get translated into `%1$@`.

Probably not for this PR, but it should not be too difficult to find out the difference between the String format and the formatted string, and compute the argument values positions in the final String.

```
func formattedString(format: String, args: [Any]) -> (final: String, argsRange: [Range])
```

**Comment by crazytonyli** [comment]
I wonder if a generic `Read more SFImage("safari")` button would suffice. That'll save us trouble in adding a link to "Visit <site> for the full post"

**Review by crazytonyli — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift:250]: Hey, @nbradbury, would you mind checking how the scenario from the PR is detected in Android?

Comment (nbradbury) [reply WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift:250]: On Android, when the post is tapped in the list view we show the excerpt in the detail view, with a link at the bottom to visit the site for the full post.

reader

**Pull request merged**